Added EWT and PB Timing for calorimeter

bechenard opened the pull request:
Added EWT support for Offline generation. There are a few additional things that needs to be done (adding ghost hits, digitization,...) once the calo digitization is finalized.

FNALbuild commented:
Hi @bechenard,
You have proposed changes to files in these packages:
- CaloMC

which require these tests: build.

@Mu2e/fnalbuild-users, @Mu2e/write have access to CI actions on [master](/Mu2e/Offline/tree/master).


:hourglass: The following tests have been triggered for 56d7f543e5c3be2cfcc816b062f9b743868bedbf: build  (Build queue is empty)


[About FNALbuild.](https://mu2ewiki.fnal.gov/wiki/Git#GitHub_Pull_Request_Procedures_and_FNALbuild) [Code review on Mu2e/Offline.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#Code_Review)

FNALbuild commented:
:memo: The latest commit by @echenard is timestamped 0:10:01.519353 (hh:mm:ss) in the future. Please check that the date and time is set correctly when creating new commits.

FNALbuild commented:
:sunny: The tests passed at ref 56d7f543e5c3be2cfcc816b062f9b743868bedbf.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :heavy_check_mark: | Merged 56d7f543e5c3be2cfcc816b062f9b743868bedbf at 85c6ef3edbe40e08bfe1e993619767b10bf1aff8 |
| scons build (prof) | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/scons.log). Build time: 13 min 14 sec |
| ceSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/ceSimReco.log) |
| transportOnly | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/transportOnly.log) |
| PS | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/PS.log) |
| g4study | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/g4study.log) |
| cosmicSimReco | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/cosmicSimReco.log) |
| rootOverlaps | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/rootOverlaps.log) |
| g4surfaceCheck | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/g4surfaceCheck.log) |
| g4test_03MT | :heavy_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/g4test_03MT.log) |
| FIXME, TODO count | :heavy_check_mark: | [TODO (0) FIXME (0) in 1 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/fixme_todo.log) |
| clang-tidy | :heavy_check_mark: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//446/console).
Build artefacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.


Logfiles may also be accessed [at this link.](https://gist.github.com/4339240b001df8cee6a18611a9ff7875)

bonventre commented on CaloMC/src/CaloShowerROMaker_module.cc:
```diff
@@ -226,9 +243,10 @@ namespace mu2e {
+              double hitTime         = hitTimeUnfolded + pbtmc.pbtime_;
+              if (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) hitTime = fmod(hitTime+mbtime_,mbtime_);
 
               if (hitTime < blindTime_ || hitTime > mbtime_ ) continue;
```
It looks like you're limiting digitization during the offspill to only the first 1695 ns. Also a few other modules like CaloDigiMaker similarly seem to use mbtime to set the waveform size

bonventre replied:
Crap x2, I forgot this was still wrong in the tracker code as well - you shouldn't use eventLength for this in CaloDigiMaker since the front end will not know this information. you should instead define a condition like blindTime that is the end of your digitization window (for onspill only) and for offspill ignore it. I'll make a PR for the tracker change so you can see what I mean.

bechenard replied:
Hopefully I fixed it. If you tell me how to test it, I'll do it.

bechenard replied:
Ok, tell em when you have it figured out and I'll do the same.

kutschke commented:
Somewhere along the line I believe that we discussed renaming  “blindtime” to something else.  I forget if we converged on an answer.  The idea was to reserve the word “blind” to refer to blinding of physics results.

On 3/17/21, 4:04 PM, "Richie Bonventre" ***@***.******@***.***>> wrote:


@bonventre commented on this pull request.

________________________________

In CaloMC/src/CaloShowerROMaker_module.cc<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=>:

>                double hitTimeUnfolded = toff_.totalTimeOffset(istep->simParticle())+istep->time();

-              double hitTime         = fmod(hitTimeUnfolded,mbtime_);

+              double hitTime         = hitTimeUnfolded + pbtmc.pbtime_;

+              if (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) hitTime = fmod(hitTime+mbtime_,mbtime_);



               if (hitTime < blindTime_ || hitTime > mbtime_ ) continue;

Crap x2, I forgot this was still wrong in the tracker code as well - you shouldn't use eventLength for this in CaloDigiMaker since the front end will not know this information. you should instead define a condition like blindTime that is the end of your digitization window (for onspill only) and for offspill ignore it. I'll make a PR for the tracker change so you can see what I mean.

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZU34MQO53R3GLIKPFTTEEKMZANCNFSM4ZJPBOSA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=G5RK-OxTVA1pK-RtsQfKsflzEampS01U1_EMGRRoMu0&e=>.

bechenard commented:
FlashTime?

On 3/17/2021 2:06 PM, Rob Kutschke wrote:
> Somewhere along the line I believe that we discussed renaming 
> “blindtime” to something else. I forget if we converged on an answer. 
> The idea was to reserve the word “blind” to refer to blinding of 
> physics results.
>
> On 3/17/21, 4:04 PM, "Richie Bonventre" ***@***.******@***.***>> wrote:
>
>
> @bonventre commented on this pull request.
>
> ________________________________
>
> In 
> CaloMC/src/CaloShowerROMaker_module.cc<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=>:
>
> > double hitTimeUnfolded = 
> toff_.totalTimeOffset(istep->simParticle())+istep->time();
>
> - double hitTime = fmod(hitTimeUnfolded,mbtime_);
>
> + double hitTime = hitTimeUnfolded + pbtmc.pbtime_;
>
> + if (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) 
> hitTime = fmod(hitTime+mbtime_,mbtime_);
>
>
>
> if (hitTime < blindTime_ || hitTime > mbtime_ ) continue;
>
> Crap x2, I forgot this was still wrong in the tracker code as well - 
> you shouldn't use eventLength for this in CaloDigiMaker since the 
> front end will not know this information. you should instead define a 
> condition like blindTime that is the end of your digitization window 
> (for onspill only) and for offspill ignore it. I'll make a PR for the 
> tracker change so you can see what I mean.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on 
> GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=>, 
> or 
> unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZU34MQO53R3GLIKPFTTEEKMZANCNFSM4ZJPBOSA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=G5RK-OxTVA1pK-RtsQfKsflzEampS01U1_EMGRRoMu0&e=>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Mu2e/Offline/pull/375#issuecomment-801443100>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABJ27JBEQHVGPWYWDJOS5GTTEEKW5ANCNFSM4ZJPBOSA>.
>

brownd1978 commented:
I recall the discussion but not the answer.  Richie, I suggest you choose a
name related to how this is used in the hardware/firmware.
Dave

On Wed, Mar 17, 2021 at 2:11 PM Bertrand Echenard ***@***.***>
wrote:

> FlashTime?
>
> On 3/17/2021 2:06 PM, Rob Kutschke wrote:
> > Somewhere along the line I believe that we discussed renaming
> > “blindtime” to something else. I forget if we converged on an answer.
> > The idea was to reserve the word “blind” to refer to blinding of
> > physics results.
> >
> > On 3/17/21, 4:04 PM, "Richie Bonventre" ***@***.******@***.***>> wrote:
> >
> >
> > @bonventre commented on this pull request.
> >
> > ________________________________
> >
> > In
> > CaloMC/src/CaloShowerROMaker_module.cc<
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=
> >:
> >
> > > double hitTimeUnfolded =
> > toff_.totalTimeOffset(istep->simParticle())+istep->time();
> >
> > - double hitTime = fmod(hitTimeUnfolded,mbtime_);
> >
> > + double hitTime = hitTimeUnfolded + pbtmc.pbtime_;
> >
> > + if (ewMarker.spillType() == EventWindowMarker::SpillType::onspill)
> > hitTime = fmod(hitTime+mbtime_,mbtime_);
> >
> >
> >
> > if (hitTime < blindTime_ || hitTime > mbtime_ ) continue;
> >
> > Crap x2, I forgot this was still wrong in the tracker code as well -
> > you shouldn't use eventLength for this in CaloDigiMaker since the
> > front end will not know this information. you should instead define a
> > condition like blindTime that is the end of your digitization window
> > (for onspill only) and for offspill ignore it. I'll make a PR for the
> > tracker change so you can see what I mean.
> >
> > —
> > You are receiving this because you are on a team that was mentioned.
> > Reply to this email directly, view it on
> > GitHub<
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=>,
>
> > or
> > unsubscribe<
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZU34MQO53R3GLIKPFTTEEKMZANCNFSM4ZJPBOSA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=G5RK-OxTVA1pK-RtsQfKsflzEampS01U1_EMGRRoMu0&e=
> >.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Mu2e/Offline/pull/375#issuecomment-801443100>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ABJ27JBEQHVGPWYWDJOS5GTTEEKW5ANCNFSM4ZJPBOSA
> >.
> >
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/375#issuecomment-801445647>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH573XWRCRXXMAP5KIABTTEELIDANCNFSM4ZJPBOSA>
> .
>


-- 
David Nathan Brown            ***@***.***
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

bechenard commented:
I'll let Richie sort it out and copy the solution for the calo.

Bertrand

On 3/17/2021 2:14 PM, David Brown wrote:
> I recall the discussion but not the answer. Richie, I suggest you choose a
> name related to how this is used in the hardware/firmware.
> Dave
>
> On Wed, Mar 17, 2021 at 2:11 PM Bertrand Echenard ***@***.***>
> wrote:
>
> > FlashTime?
> >
> > On 3/17/2021 2:06 PM, Rob Kutschke wrote:
> > > Somewhere along the line I believe that we discussed renaming
> > > “blindtime” to something else. I forget if we converged on an answer.
> > > The idea was to reserve the word “blind” to refer to blinding of
> > > physics results.
> > >
> > > On 3/17/21, 4:04 PM, "Richie Bonventre" ***@***.******@***.***>> 
> wrote:
> > >
> > >
> > > @bonventre commented on this pull request.
> > >
> > > ________________________________
> > >
> > > In
> > > CaloMC/src/CaloShowerROMaker_module.cc<
> > 
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=
> > >:
> > >
> > > > double hitTimeUnfolded =
> > > toff_.totalTimeOffset(istep->simParticle())+istep->time();
> > >
> > > - double hitTime = fmod(hitTimeUnfolded,mbtime_);
> > >
> > > + double hitTime = hitTimeUnfolded + pbtmc.pbtime_;
> > >
> > > + if (ewMarker.spillType() == EventWindowMarker::SpillType::onspill)
> > > hitTime = fmod(hitTime+mbtime_,mbtime_);
> > >
> > >
> > >
> > > if (hitTime < blindTime_ || hitTime > mbtime_ ) continue;
> > >
> > > Crap x2, I forgot this was still wrong in the tracker code as well -
> > > you shouldn't use eventLength for this in CaloDigiMaker since the
> > > front end will not know this information. you should instead define a
> > > condition like blindTime that is the end of your digitization window
> > > (for onspill only) and for offspill ignore it. I'll make a PR for the
> > > tracker change so you can see what I mean.
> > >
> > > —
> > > You are receiving this because you are on a team that was mentioned.
> > > Reply to this email directly, view it on
> > > GitHub<
> > 
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23discussion-5Fr596380838&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=161f4E9Nsf5DG6_oWNdQ2uPsNR_1cge-ls6mWAEUQ18&e=>,
> >
> > > or
> > > unsubscribe<
> > 
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZU34MQO53R3GLIKPFTTEEKMZANCNFSM4ZJPBOSA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=1qwdaU_M2XFW9jszhs1Ae66GJfBR0R0zI5nNuDOBM9c&s=G5RK-OxTVA1pK-RtsQfKsflzEampS01U1_EMGRRoMu0&e=
> > >.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/Mu2e/Offline/pull/375#issuecomment-801443100>, or
> > > unsubscribe
> > > <
> > 
> https://github.com/notifications/unsubscribe-auth/ABJ27JBEQHVGPWYWDJOS5GTTEEKW5ANCNFSM4ZJPBOSA
> > >.
> > >
> >
> > —
> > You are receiving this because you were assigned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Mu2e/Offline/pull/375#issuecomment-801445647>, or
> > unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/ABAH573XWRCRXXMAP5KIABTTEELIDANCNFSM4ZJPBOSA>
> > .
> >
>
>
> -- 
> David Nathan Brown ***@***.***
> Office Phone (510) 486-7261 Fax 495-2957
> Lawrence Berkeley National Lab
> MS 50R5008 (50-6026C) Berkeley, CA 94720
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Mu2e/Offline/pull/375#issuecomment-801447420>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABJ27JAFZJ7N34NIF5FR3ODTEELUFANCNFSM4ZJPBOSA>.
>

bonventre commented:
Currently we call the start time flashEnd but its probably more clear to just call them digitizationStart and digitizationEnd

kutschke commented:
I like digitizationStart and digitizationEnd.  Do we need to qualify them with tracker/calo/crv/mstm/extmon ? If many will be the same then we can use digitizationStart and digitizationEnd for those and a qualified name of the exceptions?


On 3/17/21, 4:24 PM, "Richie Bonventre" ***@***.******@***.***>> wrote:


Currently we call the start time flashEnd but its probably more clear to just call them digitizationStart and digitizationEnd

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23issuecomment-2D801452669&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=JuPlZhJYDAFNyqaTnv3MMRXGfvZW4edu9UU6gIEubq8&s=lFyoEKdK-tCXqRec9V9kFMhX-8YHppro0EJ1rPoVZ8k&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZXDPZVUSZWWIU7TIALTEEMXVANCNFSM4ZJPBOSA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=JuPlZhJYDAFNyqaTnv3MMRXGfvZW4edu9UU6gIEubq8&s=SASSj9nJRvWd03G7fl2owTSg3RUjnjlarhreFMulSao&e=>.

bechenard commented:
I can use CaloDigitizeStart and CaloDigitizeEnd, I like it.

Bertrand

On 3/17/2021 2:42 PM, Rob Kutschke wrote:
> I like digitizationStart and digitizationEnd. Do we need to qualify 
> them with tracker/calo/crv/mstm/extmon ? If many will be the same then 
> we can use digitizationStart and digitizationEnd for those and a 
> qualified name of the exceptions?
>
>
> On 3/17/21, 4:24 PM, "Richie Bonventre" ***@***.******@***.***>> wrote:
>
>
> Currently we call the start time flashEnd but its probably more clear 
> to just call them digitizationStart and digitizationEnd
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on 
> GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23issuecomment-2D801452669&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=JuPlZhJYDAFNyqaTnv3MMRXGfvZW4edu9UU6gIEubq8&s=lFyoEKdK-tCXqRec9V9kFMhX-8YHppro0EJ1rPoVZ8k&e=>, 
> or 
> unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZXDPZVUSZWWIU7TIALTEEMXVANCNFSM4ZJPBOSA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=JuPlZhJYDAFNyqaTnv3MMRXGfvZW4edu9UU6gIEubq8&s=SASSj9nJRvWd03G7fl2owTSg3RUjnjlarhreFMulSao&e=>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Mu2e/Offline/pull/375#issuecomment-801461922>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABJ27JGBQXACG57M4A4A63TTEEO2RANCNFSM4ZJPBOSA>.
>

gaponenko commented:
Not extmon.  It has little to do with the the timing of the beam flash 
inside the DS, so those parameters are not relevant for us.

Andrei

On Wed, 17 Mar 2021, Rob Kutschke wrote:

> I like digitizationStart and digitizationEnd.  Do we need to qualify them with tracker/calo/crv/mstm/extmon ? If many will be the same then we can use digitizationStart and digitizationEnd for those and a qualified name of the exceptions?
> 
> 
> On 3/17/21, 4:24 PM, "Richie Bonventre" ***@***.******@***.***>> wrote:
> 
> 
> Currently we call the start time flashEnd but its probably more clear to just call them digitizationStart and digitizationEnd
> 
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23issuecomment-2D801452669&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=JuPlZhJYDAFNyqaTnv3MMRXGfvZW4edu9UU6gIEubq8&s=lFyoEKdK-tCXqRec9V9kFMhX-8YHppro0EJ1rPoVZ8k&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZXDPZVUSZWWIU7TIALTEEMXVANCNFSM4ZJPBOSA&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=JuPlZhJYDAFNyqaTnv3MMRXGfvZW4edu9UU6gIEubq8&s=SASSj9nJRvWd03G7fl2owTSg3RUjnjlarhreFMulSao&e=>.
> 
> 
> -- 
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub:
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_375-23issuecomment-2D801461922&d=DwIFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=71Hc-TA7Dj0dfjGLul0zLDcBcTakklXc6KeqwOvr9fs&s=WoKuOU-4JTiQYtsNv6fAhI5XCcCMnCTVY0864OfAnxY&e=

bonventre commented:
Made a draft of the tracker version (#378), still testing that the hit ghosting is done correctly

bonventre commented on CaloMC/src/CaloShowerROMaker_module.cc:
```diff
@@ -226,11 +243,12 @@ namespace mu2e {
-
-              if (hitTime < blindTime_ || hitTime > mbtime_ ) continue;
+              double mbLength        = (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) ? mbtime_ : ewMarker.eventLength();
+              double hitTime         = fmod(hitTimeUnfolded + pbtmc.pbtime_ + mbLength, mbLength);
```
I think all the other systems have been doing no folding at all for offpsill events

kutschke replied:
That seems right.  It's a tiny effect and I doubt that we will ever care about it.  Just drop any hits/steps at times outside of the window.  Should we ever decide to care about this we will need a coordinated discussion that involves all subsystems plus whoever is defining mixing-streams for off spill.

bonventre commented:
With these changes I think calo will now have the right digi times relative to the event marker, but the change from 1695 to 1675/1700 ns windows does not seem to be implemented yet, so the end of the events will still be wrong. I'm assuming that's still planned for a future pr?

FNALbuild commented:
:hourglass: The following tests have been triggered for 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75: build  (Build queue is empty)

FNALbuild commented:
:sunny: The tests passed at 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75 at bb0e52f2e6627abe223e8adaf6fe326cead595df |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/scons.log). Build time: 24 min 38 sec |
| ceSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/ceSimReco.log) |
| g4test_03MT | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/g4test_03MT.log) |
| transportOnly | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/transportOnly.log) |
| PS | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/PS.log) |
| g4study | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/g4study.log) |
| cosmicSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/cosmicSimReco.log) |
| ceSteps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/ceSteps.log) |
| ceDigi | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/ceDigi.log) |
| muDauSteps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/muDauSteps.log) |
| ceMix | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/ceMix.log) |
| rootOverlaps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/rootOverlaps.log) |
| g4surfaceCheck | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/g4surfaceCheck.log) |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 1 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//611/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75: build  (Build queue is empty)

FNALbuild commented:
:sunny: The tests passed at 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75 at dcea32f6d7f2909bd76520513d8da121190677f2 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/scons.log). Build time: 14 min 24 sec |
| ceSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/ceSimReco.log) |
| g4test_03MT | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/g4test_03MT.log) |
| transportOnly | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/transportOnly.log) |
| POT | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/POT.log) |
| g4study | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/g4study.log) |
| cosmicSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/cosmicSimReco.log) |
| ceSteps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/ceSteps.log) |
| ceDigi | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/ceDigi.log) |
| muDauSteps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/muDauSteps.log) |
| ceMix | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/ceMix.log) |
| rootOverlaps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/rootOverlaps.log) |
| g4surfaceCheck | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/g4surfaceCheck.log) |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 1 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//632/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

FNALbuild commented:
:memo: The HEAD of `master` has changed to da4772bcb99f7b42bd401c3b15d56a211124553a. Tests are now out of date.

kutschke commented:
@FNALbuild run build tests

kutschke commented:
Adding a comment to see if that changes anything.

FNALbuild commented:
:hourglass: The following tests have been triggered for 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75: build  (Build queue has 2 jobs)

FNALbuild commented:
:sunny: The tests passed at 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75 at da4772bcb99f7b42bd401c3b15d56a211124553a |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/scons.log). Build time: 14 min 21 sec |
| ceSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/ceSimReco.log) |
| g4test_03MT | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/g4test_03MT.log) |
| transportOnly | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/transportOnly.log) |
| POT | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/POT.log) |
| g4study | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/g4study.log) |
| cosmicSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/cosmicSimReco.log) |
| ceSteps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/ceSteps.log) |
| ceDigi | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/ceDigi.log) |
| muDauSteps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/muDauSteps.log) |
| ceMix | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/ceMix.log) |
| rootOverlaps | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/rootOverlaps.log) |
| g4surfaceCheck | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/g4surfaceCheck.log) |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//653/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

FNALbuild commented:
:memo: The HEAD of `master` has changed to b7ae15e3a3f730e8fd1b09e0c6f952344d9ae23a. Tests are now out of date.

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for 8a7161aeda5f2e271bfabb3f6c03add3cbd51d75: build  (Build queue has 2 jobs)

FNALbuild commented:
:bangbang: The build test could not run due to merge conflicts. Please resolve these first and try again.
```
> git diff --check | grep -i conflict
CaloMC/src/CaloDigiMaker_module.cc:58: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:71: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:86: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:91: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:93: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:96: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:118: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:125: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:132: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:194: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:205: leftover conflict marker
CaloMC/src/CaloDigiMaker_module.cc:213: leftover conflict marker
```

kutschke commented:
Bertrand: one of the recently merged PRs has created a conflict with your branch.  It looks like one file only:

CaloMC/src/CaloDigiMaker_module.cc

FNALbuild commented:
:memo: The HEAD of `master` has changed to bb35bc296b387a993ef766f47eee1279e4dc8f14. Tests are now out of date.

bonventre commented on CaloMC/src/CaloDigiMaker_module.cc:
```diff
@@ -140,31 +144,35 @@ namespace mu2e {
+      double mbtime    = (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) ?  accPar->deBuncherPeriod : ewMarker.eventLength();      
       int nWaveforms   = calorimeter_->nCrystal()*calorimeter_->caloInfo().getInt("nSiPMPerCrystal");
-      int waveformSize = (mbtime_ - blindTime_ + endTimeBuffer_) / digiSampling_; 
+      int waveformSize = (mbtime - blindTime_ + endTimeBuffer_) / digiSampling_; 
```
Ok reading through it again I think it will handle the end of the event properly, but you have fixed the digitization length to be 1695-blindTime+endTimeBuffer. It might be more straightforward to have digitizationStart and digitizationEnd instead as we do in the tracker

kutschke replied:
When Richie made that that change ( to  digitizationStart and digitizationEnd ) there was a consensus the new way is the right way to think about it.  Also people objected to the word "blind" as a distraction because it suggested signal blinding.

bechenard commented:
Hi guys,

I rebased my code and fixed the conflict, but I'm afraid I haven't had time to implement all suggestions to mimic the tracker structure. My understanding from previous comments from Richie is that it should work. I tested it with OnSpill CE and it runs fine. I also processed the code with OffSpill digitization and it ran, but I haven't heard from Dave about a begin-to-end test so there are no guarantees. 

I'll let you have a look and decide what needs to be done next, but this should be good enough for MDC2020.

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4: build  (Build queue has 2 jobs)

FNALbuild commented:
:sunny: The tests passed at ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4 at 4ae3e29fcebcc9f5853b9541acc4c3428f0a0cf1 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/scons.log). Build time: 13 min 31 sec |
| ceSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/ceSimReco.log) |
| g4test_03MT | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/g4test_03MT.log) |
| transportOnly | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/transportOnly.log) |
| POT | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/POT.log) |
| g4study | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/g4study.log) |
| cosmicSimReco | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/cosmicSimReco.log) |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/ceSteps.log) |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/ceDigi.log) |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/muDauSteps.log) |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/ceMix.log) |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/rootOverlaps.log) |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/g4surfaceCheck.log) |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//736/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
Thanks Bertrand.  This now tests cleanly.  Dave will look after the merge.

FNALbuild commented:
:memo: The HEAD of `master` has changed to e8a59dc4467c617f000be4fbfa5ac02b995e8b86. Tests are now out of date.

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4: build  (Build queue has 1 jobs)

FNALbuild commented:
:umbrella: The tests failed for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4 at e8a59dc4467c617f000be4fbfa5ac02b995e8b86 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/scons.log). Build time: 13 min 51 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/ceSteps.log)  |
| ceDigi | :stopwatch: :x: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/ceDigi.log) Timed out. |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//743/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4: build  (Build queue has 1 jobs)

FNALbuild commented:
:sunny: The tests passed at ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4 at 50e389c2dc164b929052faf041e2f15f3ff766ed |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/scons.log). Build time: 27 min 03 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//747/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

FNALbuild commented:
:memo: The HEAD of `master` has changed to 49305e0620a728af8f2ddfdcb1a9c6b4734aecb1. Tests are now out of date.

kutschke commented:
@FNALbuild run build tests

Expect ceMix to fail until the input files are updated.

FNALbuild commented:
:hourglass: The following tests have been triggered for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4: build  (Build queue has 3 jobs)

FNALbuild commented:
:umbrella: The tests failed for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4 at 49305e0620a728af8f2ddfdcb1a9c6b4734aecb1 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/scons.log). Build time: 14 min 33 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/muDauSteps.log)  |
| ceMix | :x: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/ceMix.log) Return Code 1. |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//750/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
See if Dave's fix to Production clears the ceMix issue

@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4: build  (Build queue has 3 jobs)

FNALbuild commented:
:umbrella: The tests failed for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4 at 49305e0620a728af8f2ddfdcb1a9c6b4734aecb1 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/scons.log). Build time: 13 min 45 sec |
| ceSimReco | :x: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/ceSimReco.log) Return Code 65. |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//755/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4: build  (Build queue is empty)

FNALbuild commented:
:sunny: The tests passed at ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged ab085fe3e5beb3f8ca72e97bc1765a9a33d919d4 at 8a68c972c83804b8f69f82576ce93ae3664853cf |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/scons.log). Build time: 13 min 00 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//759/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.
